tests/unit_tests/test_CRISPRessoCORE.py

```diff
@@ -93,6 +93,105 @@ def test_get_consensus_alignment_from_pairs():
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_deletion():
+    quant_window_coordinates = '10-20_35-40'
+    # represents a 7bp deletion in the middle
+    s1inds = list(range(23)) + [22, 22, 22, 22, 22, 22, 22] + list(range(23, 34))
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [*list(range(10, 21)), *list(range(35-7, 41-7))]
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_deletion_modified():
+    quant_window_coordinates = '10-25_35-40'
+    # represents a 7bp deletion in the middle, where part of the QW is deleted
+    # [0, 1, 3, 4, ... , 21, 22, 22, 22, 22, 22, 22, 22, 22, 23, 24, ... , 33]
+    s1inds = list(range(23)) + [22, 22, 22, 22, 22, 22, 22] + list(range(23, 34))
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [*list(range(10, 23)), *list(range(35-7, 41-7))]
+
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_deletion_end_modified(): 
+    # 5 bp deletion at end of 20 bp sequence
+    quant_window_coordinates = '1-5_10-20'
+    s1inds = [*list(range(16)), *[15, 15, 15, 15, 15]]
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [*list(range(1, 6)), *list(range(10, 16))]
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_insertion_and_deletion():
+    # 5 bp deletion and 5 bp insertion
+    quant_window_coordinates = '1-5_10-20'
+    s1inds = [0, 1, 2, 3, 4, 5, 5, 5, 5, 5, 5, 6, 7, 8, 9, 15]
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [*list(range(1, 6)), *[6, 7, 8, 9, 15]]
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_insertion_and_deletion_modified():
+    # 5 bp deletion and 5 bp insertion with modified QW
+    pytest.xfail('Not implemented yet')
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_insertion_across_qw():
+    # 6 bp insertion in middle of 4 bp sequence
+    quant_window_coordinates = '1-4'
+    s1inds = [0,1,2,9,10]
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [1,2,9,10]
+
+def test_get_cloned_include_idxs_from_quant_window_coordinates_deletion_entire_qw():
+    # 5 bp deletion of entire qw
+    quant_window_coordinates = '1-4_7-10'
+    s1inds = [0, 1, 2, 3, 4, 5, 6, 6, 6, 6, 6]
+    assert CRISPRessoCORE.get_cloned_include_idxs_from_quant_window_coordinates(quant_window_coordinates, s1inds) == [1, 2, 3, 4]
+
+# NOTE: include_idxs should be exactly the numbers as specificed in the string, inclusively
+# For example: 20-30_175-185 --> [20, 21, ..., 30, 175, 176, ..., 185]
+    
+# BUG: you can't include the last base in the QW or you get an out of range error -- this is extant beyond the function tested here.
+
```

Comment:
Could this be addressed in the PR? Or would it be better addressed in a separate PR?

Reply:
PS, what would Uncle Bob think of this comment? ;)

Reply:
I left this as a reminder. It worked! haha

So, this touches on a key issue: QWC's are (supposedly) 1-indexed. All of these changes assume that downstream calculations correctly utilize 1-index-based coordinates as input by the user, i.e. the user inputs `'1-5' --> include_idxs == [1, 2, 3, 4, 5]`. The test cases verify that these indexes are selected correctly.

So, I think the only necessary change to get the last char in the sequence would be to append one more index onto s1inds before we slice it.

Reply:
I think technically they [should be 0-indexed](https://github.com/pinellolab/CRISPResso2?tab=readme-ov-file#quantification-window-parameters) (but a value of 0 will disable the filter).

Reply:
Per the documentation "Indexes are 0-based, meaning that the first nucleotide is position 0." Thanks @kclem. So, there is no last bp bug.

However, including 0 in the qwc throws a `BadParameterException`, I'm assuming because 0 disables the filter. Should we keep this behavior as is?

Reply:
@mbowcut2 I think it should behave such that if you have `-qwc 0-10_15-22,0` the first amplicon will have quantification windows `0-10` and `15-22` and then the second amplicon will have the quantification windo coordinates disabled (and therefore be set by where the sgRNA aligns).

Also, `-qwc 0` will outright disable it.

Reply:
@Colelyman update here: 

First, `-qwc` accepts 0 as part of a range and works correctly as long as `--exclude_bp_from_left` is set to 0 (the default is 15). 

Second, `-qwc` doesn't accept `0` as an argument. But it can be used by `-w` e.g. `-w 0`. It seems that `-qwc` supersedes other quantification window options, and so `-w 0 -qwc 10-20` would give you a window of 10-20.

Reply:
@Colelyman okay, it works now.